### PR TITLE
feat: support `|&` pipelines

### LINKED ISSUE #12: Support `|&` pipelines
This pipes stdout & stderr to stdin instead of only stdout to stdin.

### PR DESCRIPTION
**Pull request by dsherret**
These pipelines map stdout and stderr to stdin of the next command.

Closes #12.

### CODE REVIEW

File: src/shell.rs [hunk 1]
```diff
@@ -352,7 +365,7 @@ async fn execute_command(
       .envs(state.env_vars())
       .stdout(stdout.into_raw())
       .stdin(stdin.into_raw())
-      .stderr(Stdio::inherit())
+      .stderr(stderr.clone().into_raw())
```
Comment (dsherret) [inline src/shell.rs:368]: Not a big deal because previously stderr was always inherited from the process anyway.

File: src/test.rs [hunk 2]
```diff
@@ -240,6 +240,35 @@ pub async fn pipeline() {
...
+    .assert_stderr("2\n1\n")
+    .run()
+    .await;
+
+  // |& pipeline should still pipe stdout
+  TestBuilder::new()
+    .command(r#"echo 1 |& deno eval 'await Deno.stdin.readable.pipeTo(Deno.stdout.writable)'"#)
+    .assert_stdout("1\n")
+    .run()
+    .await;
```
Comment (bartlomieju) [inline src/test.rs:271]: Nice tests!

**Review by bartlomieju — APPROVED** [review]
LGTM, I'm still bothered by `char` function which is colored as a keyword 😅

**Comment by dsherret** [comment]
> LGTM, I'm still bothered by `char` function which is colored as a keyword 😅

One day I will change it. Right now it's still aligned with nom.

**Pull request merged**